torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp

```diff
@@ -17,171 +18,1092 @@ namespace fuser {
+            pos_after_outer != inner_pos) {
+          // Skip broadcast axes as they must not have been concretized in the
+          // reference. We remove dimensions that underwent a concretization as
+          // well as the dimensions to the left of that.
+          continue;
+        }
+        break;
+      }
+
+      if (inner_pos != pos_after_outer) {
+        // Nothing to the left of inner could be continuous.
+        //
+        // Clear left of inner
+        clear_left_of(find_inner_it);
+      }
+
+      auto& inner_mapping = getMappedExtent(split->inner());
+      if (inner_mapping.isZero()) {
+        clear_left_of(find_inner_it + 1);
+        continue;
+      }
+
+      ids[inner_pos] = split->in();
+
+      if (find_outer_it != ids.end() && find_inner_it != ids.end()) {
+        // Both dimensions map, inner dimension maps fully. For more context see
+        // the comment:
+        //  projectIdToRFactor
+        //    if (find_outer_it != ids.end() && find_inner_it != ids.end() &&
+        //      !hasPartialExtent(merge->inner())) {
+        //      Comment here.
+
+        // Remove outer
+        ids.erase(find_outer_it);
+
+        const auto inner_numerator = inner_mapping.getNumerator();
+        const auto inner_denominator = inner_mapping.getDenominator();
+
+        // Is the inner dimension is fully mapped
+        auto inner_is_fully_mapped = IrBuilder::eqExpr(
+            SimplifyingIrBuilder::divExpr(inner_numerator, inner_denominator),
+            commonOrConstExtent(ca_map_, split->inner()));
+
+        // Divisibility checks are done later, simply propagate fractional
+        // values through the graph.
+
+        // Always propagate inner dimension to in backward through split
+        auto in_mapping = inner_mapping;
+        // If inner is fully mapped propagate the outer mapping as well
+        const auto outer_mapping = getMappedExtent(split->outer());
+        in_mapping.maybeMul(inner_is_fully_mapped, outer_mapping);
+        projected_extent_[split->in()] = in_mapping;
+      } else {
+        // Only inner maps
+        projected_extent_[split->in()] = getMappedExtent(split->inner());
+        // Clear to the left of inner in since only inner maps
+        ids.erase(ids.begin(), ids.begin() + inner_pos);
+      }
+    } else if (const Merge* merge = dynamic_cast<const Merge*>(expr)) {
+      auto find_out_it = std::find(ids.begin(), ids.end(), merge->out());
+      if (find_out_it == ids.end()) {
+        continue;
+      }
+
+      auto out_pos = std::distance(ids.begin(), find_out_it);
+
+      // Map inner and outer dimensions. If outer doesn't map fully this will be
+      // found during evaluation. Don't have to worry about it here.
+      ids[out_pos] = merge->outer();
+      ids.insert(ids.begin() + out_pos + 1, merge->inner());
+
+      auto out_mapping = getMappedExtent(merge->out());
+      if (out_mapping.isZero()) {
+        ids.erase(ids.begin(), ids.begin() + out_pos + 1);
```

Comment:
Shouldn't this be `ids.erase(ids.begin(), ids.begin() + out_pos + 2)`? The inner ID was inserted just before this part of the code, and if the out is zero, both the outer and inner IDs should be discarded, not?

Reply:
Moved the insertion to after this block, thanks.